src/Components/Aspire.Azure.AI.Inference/AspireAzureAIInferenceExtensions.cs

```diff
@@ -0,0 +1,205 @@
+            {
+                if (settings.Endpoint is null)
+                {
+                    throw new InvalidOperationException($"A ChatCompletionsClient could not be configured. Ensure valid connection information was provided in 'ConnectionStrings:{connectionName}' or specify a '{nameof(ChatCompletionsClientSettings.Endpoint)}' and optionally a '{nameof(ChatCompletionsClientSettings.Key)}' in the '{configurationSectionName}' configuration section.");
+                }
+                else
+                {
+                    // Connect to Azure AI Foundry using key auth
+                    if (!string.IsNullOrEmpty(settings.Key))
+                    {
+                        var credential = new AzureKeyCredential(settings.Key);
+                        return new ChatCompletionsClient(settings.Endpoint, credential, options);
+                    }
+                    else
+                    {
+                        return new ChatCompletionsClient(settings.Endpoint, settings.TokenCredential ?? new DefaultAzureCredential(), options);
+                    }
+                }
+            });
+        }
+
+        protected override void BindClientOptionsToConfiguration(IAzureClientBuilder<ChatCompletionsClient, AzureAIInferenceClientOptions> clientBuilder, IConfiguration configuration)
+        {
+#pragma warning disable IDE0200 // Remove unnecessary lambda expression - needed so the ConfigBinder Source Generator works
+            clientBuilder.ConfigureOptions(options => configuration.Bind(options));
+#pragma warning restore IDE0200
+        }
+
+        protected override void BindSettingsToConfiguration(ChatCompletionsClientSettings settings, IConfiguration configuration)
+            => configuration.Bind(settings);
+
+        protected override IHealthCheck CreateHealthCheck(ChatCompletionsClient client, ChatCompletionsClientSettings settings)
+            => throw new NotImplementedException();
+
+        protected override bool GetHealthCheckEnabled(ChatCompletionsClientSettings settings)
+            => false;
+
+        protected override bool GetMetricsEnabled(ChatCompletionsClientSettings settings)
+            => !settings.DisableMetrics;
+
+        protected override TokenCredential? GetTokenCredential(ChatCompletionsClientSettings settings)
+            => settings.TokenCredential;
+
+        protected override bool GetTracingEnabled(ChatCompletionsClientSettings settings)
+            => !settings.DisableTracing;
+    }
+
+    /// <summary>
+    /// Creates a <see cref="IChatClient"/> from the <see cref="ChatCompletionsClient"/> registered in the service collection.
+    /// </summary>
+    /// <param name="builder">An <see cref="AspireChatCompletionsClientBuilder" />.</param>
+    /// <returns></returns>
+    public static ChatClientBuilder AddChatClient(this AspireChatCompletionsClientBuilder builder)
```

Comment:
Should this be named `AddIChatClient` to align to `.AsIChatClient()` in the underlying MEAI library?